### PR TITLE
Drop support for python 3.6

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

Drop support for python 3.6.

## How is this patch tested?

Confirming all the existing checks pass.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Support for Python 3.6 has been dropped.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
We also need to update our internal Jenkins jobs to use python 3.7.

File: mlflow/__init__.py [hunk 1]
```diff
@@ -100,24 +98,25 @@
...
+#         "an upcoming release. At that point, existing Python {dep_ver} workflows "
+#         "that use MLflow will continue to work without modification, but Python {dep_ver} "
+#         "users will no longer get access to the latest MLflow features and bugfixes. "
+#         "We recommend that you upgrade to Python {min_ver} or newer.".format(
+#             dep_ver=".".join(map(str, _deprecated_version)),
+#             min_ver=".".join(map(str, _min_supported_version)),
+#         ),
+#         FutureWarning,
+#         stacklevel=2,
+#     )
```
Comment (harupy) [inline mlflow/__init__.py:119]: We'll reuse this block in the future.

File: setup.py [hunk 2]
```diff
@@ -129,10 +129,10 @@ def package_files(directory):
     else open("README_SKINNY.rst").read() + open("README.rst").read(),
     long_description_content_type="text/x-rst",
     license="Apache License 2.0",
-    classifiers=["Intended Audience :: Developers", "Programming Language :: Python :: 3.6"],
+    classifiers=["Intended Audience :: Developers", "Programming Language :: Python :: 3.7"],
     keywords="ml ai databricks",
     url="https://mlflow.org/",
-    python_requires=">=3.6",
+    python_requires=">=3.7",
```
Comment (harupy) [inline setup.py:135]: 🎉

**Comment by harupy** [comment]
ref: https://realpython.com/python37-new-features/

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

**Pull request merged**